pkgs/development/python-modules/conda/default.nix

```diff
@@ -1,38 +1,62 @@
-{ lib
-, buildPythonPackage
-, pythonAtLeast
-, fetchPypi
-, pycosat
-, requests
-, ruamel-yaml
-, isPy3k
-, enum34
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hatchling,
+  hatch-vcs,
+  archspec,
+  conda-libmamba-solver,
+  conda-package-handling,
+  distro,
+  jsonpatch,
+  packaging,
+  platformdirs,
+  pluggy,
+  pycosat,
+  requests,
+  ruamel-yaml,
+  tqdm,
+  truststore,
+  ...
 }:
-
 # Note: this installs conda as a library. The application cannot be used.
 # This is likely therefore NOT what you're looking for.
-
 buildPythonPackage rec {
   pname = "conda";
-  version = "4.3.16";
-  format = "setuptools";
+  version = "24.1.2";
+  format = "pyproject";
 
-  # this is a very outdated version of conda that isn't compatible with python 3.10+
-  disabled = pythonAtLeast "3.10";
-
-  src = fetchPypi {
+  src = fetchFromGitHub {
     inherit pname version;
-    sha256 = "a91ef821343dea3ba9670f3d10b36c1ace4f4c36d70c175d8fc8886e94285953";
+    owner = "conda";
+    repo = "conda";
+    rev = version;
+    sha256 = "sha256-L/Y7Bb3R5YqXbjTN4CRPFnkgymVLrxuFmjVzpvt28dE=";
   };
 
-  propagatedBuildInputs = [ pycosat requests ruamel-yaml ] ++ lib.optional (!isPy3k) enum34;
+  buildInputs = [hatchling hatch-vcs];
+  propagatedBuildInputs = [archspec conda-libmamba-solver conda-package-handling distro jsonpatch packaging platformdirs pluggy pycosat requests ruamel-yaml tqdm truststore];
+
+  patches = [./0001-conda_exe.patch];
+  # menuinst is a dependency that is only needed on Windows
+  postPatch = ''
+    sed -i ./pyproject.toml -e '/menuinst/d'
+  '';
+  makeWrapperArgs = [
+    "--set CONDA_EXE ${placeholder "out"}/bin/conda"
+    ''--set CONDA_ENVS_PATH "~/.conda/envs/"''
+    ''--set CONDA_PKGS_DIRS "~/.conda/pkgs/"''
```

Comment:
1. Why not `--set-default`?
2. Why not allow overriding via `.override`?

Reply:
1. I didn't know that was an option. I'll look into it.
2. I didn't think that was possible. I'll have look in to doing this.